crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -0,0 +1,360 @@
+    /// [`RpoFalcon512`][crate::account::auth::RpoFalcon512] module.
+    ///
+    /// Internal value holds the storage index where the public key for the RpoFalcon512
+    /// authentication scheme is stored.
+    RpoFalcon512(u8),
+    /// A non-standard, custom interface which exposes the contained procedures.
+    ///
+    /// Custom interface holds procedures which are not part of some standard interface which is
+    /// used by this account. Each custom interface holds procedures with the same storage offset.
+    Custom(Vec<AccountProcedureInfo>),
+}
+
+impl AccountComponentInterface {
+    /// Creates a vector of [AccountComponentInterface] instances. This vector specifies the
+    /// components which were used to create an account with the provided procedures info array.
+    pub fn from_procedures(procedures: &[AccountProcedureInfo]) -> Vec<Self> {
+        let mut component_interface_vec = Vec::new();
+
+        let mut procedures: BTreeMap<_, _> = procedures
+            .iter()
+            .map(|procedure_info| (*procedure_info.mast_root(), procedure_info))
+            .collect();
+
+        // Basic Wallet
+        // ------------------------------------------------------------------------------------------------
+
+        if basic_wallet_library()
+            .mast_forest()
+            .procedure_digests()
+            .all(|proc_digest| procedures.contains_key(&proc_digest))
+        {
+            basic_wallet_library().mast_forest().procedure_digests().for_each(
+                |component_procedure| {
+                    procedures.remove(&component_procedure);
+                },
+            );
+
+            component_interface_vec.push(AccountComponentInterface::BasicWallet);
+        }
+
+        // Basic Fungible Faucet
+        // ------------------------------------------------------------------------------------------------
+
+        if basic_fungible_faucet_library()
+            .mast_forest()
+            .procedure_digests()
+            .all(|proc_digest| procedures.contains_key(&proc_digest))
+        {
+            basic_fungible_faucet_library().mast_forest().procedure_digests().for_each(
+                |component_procedure| {
+                    procedures.remove(&component_procedure);
+                },
+            );
+
+            component_interface_vec.push(AccountComponentInterface::BasicFungibleFaucet);
+        }
+
+        // RPO Falcon 512
+        // ------------------------------------------------------------------------------------------------
+
+        let rpo_falcon_proc = rpo_falcon_512_library()
+            .mast_forest()
+            .procedure_digests()
+            .next()
+            .expect("rpo falcon 512 component should export exactly one procedure");
+
+        if let Some(proc_info) = procedures.remove(&rpo_falcon_proc) {
+            component_interface_vec
+                .push(AccountComponentInterface::RpoFalcon512(proc_info.storage_offset()));
+        }
```

Comment:
I still think encapsulating this code closer to where we specify well-known account components would be cleaner. But let's leave this for a follow-up PR.

Reply:
Created an issue for this comment and for the comments above: https://github.com/0xPolygonMiden/miden-base/issues/1204